### DISCUSSION TITLE
Data Parallel - Managing State Variables

### DISCUSSION THREAD

**Discussion by peterdavidfagan** [google/flax#3264] [Q&A]
Hi Flax Community,

I am getting started with transforming my flax training pipeline into a format that can run across multiple devices. I've been reading the following [guide](https://flax.readthedocs.io/en/latest/guides/flax_on_pjit.html) to accomplish this. I had some clarifying questions that I was hoping to ask (and post here in case the answer to these are also useful to other users).

**Managing State Variables**
For layers that maintain variables such as `nn.BatchNorm` ([API docs](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/_autosummary/flax.linen.BatchNorm.html)), is it sufficient to pass the logical axis name to the `axis_name` parameter for state variables to be correctly tracked when training across multiple devices? The API docs for BatchNorm reference pmap but don't mention partitioning with jit. Does this parameter also apply to partitioning with jit as outlined in the guide? I presume yes, but I haven't delved into the codebase for jit yet to verify.
